pkg/cmd/issue/edit/edit.go

```diff
@@ -72,7 +72,19 @@ func NewCmdEdit(f *cmdutil.Factory, runF func(*EditOptions) error) *cobra.Comman
 			// support `-R, --repo` override
 			opts.BaseRepo = f.BaseRepo
 
-			opts.SelectorArgs = args
+			issueNumbers, baseRepo, err := shared.ParseIssuesFromArgs(args)
```

Comment:
This is really where the magic happens, we move arg parsing into the pre-run, which means we can inject the correct issue numbers and base repo.

Since we now have the base repo, we are able to construct a featuredetection.Detector early to address things like https://github.com/cli/cli/pull/10785/files#r2046543890